src/jltypes.c

```diff
@@ -1415,6 +1431,11 @@ static jl_value_t *inst_datatype_inner(jl_datatype_t *dt, jl_svec_t *p, jl_value
                 if (p) jl_gc_wb(p, tw);
             }
         }
+        if (tn == jl_type_typename && jl_is_datatype(iparams[0]) && ((jl_datatype_t*)iparams[0])->name == jl_type_typename &&
+            jl_tparam0(iparams[0]) == jl_bottom_type) {
+            // normalize Type{Type{Union{}}} to Type{TypeofBottom}
+            iparams[0] = (jl_value_t*)jl_typeofbottom_type;
```

Comment:
Is this valid, even though you are changing the kind of the type for subtyping?

Reply:
We are already doing this via the cache:
```
julia> Type{Type{Union{}}}
Type{Core.TypeofBottom}
```
and we have tests that depend on it. `TypeofBottom` is also a DataType as well.